### PR TITLE
GAWB-2064: filter and sort workspaces when first loaded.

### PR DESCRIPTION
**Pull request by rushtong**
Addresses https://broadinstitute.atlassian.net/browse/GAWB-2064

---

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [x] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by MattPutnam** [comment]
Use components/WorkspaceSelector instead.

**Comment by rushtong** [comment]
I looked into that, but the `WorkspaceSelector` component complicates
things because it changes what's stored in state as compared to what
the rest of the page wants to use. I'm very hesitant to refactor this
whole page to use that component.

On Tue, Sep 26, 2017 at 1:08 PM, Matt Putnam <notifications@github.com> wrote:
> Use components/WorkspaceSelector instead.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.



-- 
Gregory Rushton
Principal Software Engineer
Data Sciences Platform
Broad Institute of MIT and Harvard
75 Ames Street
Cambridge, MA 02142
cell: 781.710.1876

**Comment by rushtong** [comment]
@MattPutnam - I looked into that, but the `WorkspaceSelector` component complicates things because it changes what's stored in state as compared to what the rest of the page wants to use. I'm very hesitant to refactor this whole page to use that component.

The other issue I had was trying to figure out how to make the first workspace auto-selected. The current code does that, but I don't know how to adapt`WorkspaceSelector` to do the same thing.

**Comment by MattPutnam** [comment]
I'm not sure what you mean. The page stores the selected workspace in state, and WorkspaceSelector takes an `:on-select` function that you would use to put the selected workspace into state. It should look almost exactly like how it's being used already (in exporter.cljs).

**Comment by rushtong** [comment]
Thanks @MattPutnam - finally got past my mental block on using that component.

File: src/cljs/main/broadfcui/page/method_repo/method_config_importer.cljs [hunk 1]
```diff
@@ -20,6 +21,7 @@
    [broadfcui.page.method-repo.methods-configs-acl :as mca]
    [broadfcui.page.method-repo.synchronize :as mr-sync]
    [broadfcui.page.method-repo.redactor :refer [Redactor]]
+   [broadfcui.page.workspace.workspace-common :as ws-common]
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/method_repo/method_config_importer.cljs]: Unused

File: src/cljs/main/broadfcui/page/method_repo/method_config_importer.cljs [hunk 2]
```diff
@@ -81,20 +83,10 @@
...
-                    (if success?
-                      (let [ws-list (get-parsed-response)]
-                        (swap! state assoc :workspaces-list ws-list :selected-workspace (first ws-list)))
-                      (swap! state assoc :error status-text)))})))
-   :render
+  {:render
    (fn [{:keys [props state locals refs]}]
      (let [{:keys [workspace-id entity perform-copy]} props
-           {:keys [workspaces-list]} @state]
+           {:keys [workspaces]} @state]
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/method_repo/method_config_importer.cljs]: Unused

**Review by MattPutnam — APPROVED** [review]
Is your IDE configured to show unused items?

**Comment by rushtong** [comment]
Definitely not set up for unused items - will look into that.

**Review by davidangb — APPROVED** [review]
:+1: quick test shows this working for me locally.

**Pull request merged**